src/Resolution.ts

```diff
@@ -515,26 +518,32 @@ export default class Resolution {
    */
   async isAvailable(domain: string): Promise<Boolean> {
     domain = prepareAndValidateDomain(domain);
-    const method = this.getNamingMethodOrThrow(domain);
-    return method.isAvailable(domain);
+    const methods = this.getNamingMethods(domain);
+    return this.callForDomain(domain, methods, 'isAvailable', [domain]);
   }
 
   /**
    * @returns Produces a namehash from supported naming service in hex format with 0x prefix.
    * Corresponds to ERC721 token id in case of Ethereum based naming service like UNS.
+   * @param namingService TODO!
    * @param domain domain name to be converted
    * @param options formatting options
    * @throws [[ResolutionError]] with UnsupportedDomain error code if domain extension is unknown
    */
   namehash(
+    namingService: NamingServiceName,
     domain: string,
     options: NamehashOptions = NamehashOptionsDefault,
```

Comment:
I think we need to be more open to zilliqa deprecation here: I think long term zilliqa should be removed, so can we make it more explicit required option like `znsHash: boolean`. It will allow us to make it optional in the future so that we only support modern domains without breaking changes.

Reply:
Can we add this parameter to `NamehashOptions`? This way it should be easier to deprecate it without changing the method signature

Reply:
Agreed. What do you think about the `childhash` method, btw? Should we refactor it the same way for consistency? (It got the `namingService` parameter a long time ago.)

Reply:
Well, after giving it another thought, introducing this as a separate argument should be safer for TypeScript users who will update their code. Interface change communicates the need to update the code more clearly. Without the breaking change, there's a high chance that users supporting both UNS and ZNS will get an invalid behaviour.